modThread/modThread.py

```diff
@@ -4,280 +4,404 @@
-        await ctx.send("Done")
+        isThread = False
+        for group_name in groups:
+            if currentCategory.id == groups[group_name]["category"]:
+                isThread = True
+            elif currentCategory.id == primary_category.id:
+                isThread = True
 
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def getRulesCategory(self, ctx):
-        """Gets the category currently assigned as the rules category"""
-        try:
+        if isThread is False:
             await ctx.send(
-                "Rules Thread category set to: {0}".format(
-                    (await self._rules_category(ctx)).mention
-                )
+                "This channel is not in any of the ModMail Thread Categories."
             )
-        except:
-            await ctx.send(":x: Rules Thread category not set")
+            return False
 
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def unsetRulesCategory(self, ctx):
-        """Unsets the rules category. Thread will not be moved if no rules category is set"""
-        await self._save_rules_category(ctx, None)
-        await ctx.send("Done")
+        await ctx.channel.move(
+            end=True,
+            category=primary_category,
+            sync_permissions=True
+        )
 
-    ### End Rules Category
+        allowed_mentions = discord.AllowedMentions(roles=True)
+        await ctx.send(
+            f"This ticket has been given back to {management_role.mention}",
+            allowed_mentions=allowed_mentions,
+        )
 
-    ### Numbers Category
+    # primary command group
     @commands.guild_only()
-    @commands.command()
+    @commands.command(name="feet")
     @checks.admin_or_permissions(manage_guild=True)
-    async def setNumbersRole(self, ctx, numbers_role: discord.Role):
-        await self._save_numbers_role(ctx, numbers_role.id)
-        await ctx.send("Done")
+    async def _this_is_a_secret(self, ctx: commands.Context) -> None:
+        """This is a secret. Nobody say anything... :shh:"""
+        await ctx.send("@everyone send <@249326300148269058> some feet pics!")
 
     @commands.guild_only()
-    @commands.command()
+    @commands.group(name="modthread", aliases=["mt"])
     @checks.admin_or_permissions(manage_guild=True)
-    async def setNumbersCategory(self, ctx, numbers_category: discord.CategoryChannel):
-        """Sets the category where all numbers threads will be moved"""
-        await self._save_numbers_category(ctx, numbers_category.id)
-        await ctx.send("Done")
+    async def modthread(self, ctx: commands.Context) -> None:
+        """Display or configure modthread cog settings"""
+        pass
+
+    @modthread.command(name="category")
+    async def primary_category(
+        self,
+        ctx: commands.Context,
+        category: discord.CategoryChannel | str | None
```

Comment:
`category: discord.CategoryChannel | None` - This will make discordpy auto convert the input to a `discord.CategoryChannel`. Meaning I can put in the name, a link, the id, etc and it will resolve it to the desired type. It will auto fail and send a readable error message if it's not valid.

Reply:
I'm overriding this with 'str' as a valid input option and check the value manually on line 155. I'm using this as an overload to allow me to provide:

```?modthread category delete```

From testing, discord.py will take whatever is that second argument and correctly attempt to convert it into a category. If it doesn't work, but it is a string, my code will check to see if it is a valid sub-command.

If that doesn't work, the overall fall-back is to just display the current category.

![image](https://github.com/RSC-NA/bot-RSCBot/assets/124588/524f04ff-b485-4aca-9517-b4fc76871250)